packages/react-tooltip/src/components/Tooltip/useTooltip.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import { makeMergeProps, resolveShorthandProps, useMergedRefs } from '@fluentui/react-utilities';
+import { usePopper } from '@fluentui/react-positioning';
+import { useTheme } from '@fluentui/react-theme-provider';
+import { TooltipProps, TooltipShorthandProps, TooltipState } from './Tooltip.types';
+import { arrowHeight, tooltipBorderRadius } from './useTooltipStyles';
+
+/**
+ * Names of the shorthand properties in TooltipProps
+ * {@docCategory Tooltip}
+ */
+export const tooltipShorthandProps: TooltipShorthandProps[] = ['arrow'];
+
+const mergeProps = makeMergeProps<TooltipState>({ deepMerge: tooltipShorthandProps });
+
+/**
+ * Create the state required to render Tooltip.
+ *
+ * The returned state can be modified with hooks such as useTooltipStyles,
+ * before being passed to renderTooltip.
+ *
+ * @param props - props from this instance of Tooltip
+ * @param ref - reference to root HTMLElement of Tooltip
+ * @param defaultProps - (optional) default prop values provided by the implementing type
+ *
+ * {@docCategory Tooltip}
+ */
+export const useTooltip = (
+  props: TooltipProps,
+  ref: React.Ref<HTMLElement>,
+  defaultProps?: TooltipProps,
+): TooltipState => {
+  const [target, setTarget] = React.useState<HTMLElement | null>(null);
+
+  const state = mergeProps(
+    {
+      ref,
+      as: 'div',
+      role: 'tooltip',
+      position: 'above',
+      align: 'center',
+      offset: 4,
+      arrow: { as: 'div', children: '' },
+      visible: !!target,
+    },
+    defaultProps && resolveShorthandProps(defaultProps, tooltipShorthandProps),
+    resolveShorthandProps(props, tooltipShorthandProps),
+  );
+
+  if (state.noArrow) {
+    state.arrow = undefined;
+  }
+
+  React.useImperativeHandle(
+    state.componentRef,
+    () => ({
+      show: t => setTarget(t),
+      hide: () => setTarget(null),
+    }),
+    [],
```

Comment:
is it intentional that `setTarget` is not a dep ?

Reply:
`React.useState` ensures that the setter function it returns is stable between renders.

https://reactjs.org/docs/hooks-reference.html#usestate
> React guarantees that setState function identity is stable and won’t change on re-renders. This is why it’s safe to omit from the useEffect or useCallback dependency list.